PBLV5/Core/Src/CurrentMeasurement.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+ * CurrentMeasurement.cpp
+ *
+ *  Created on: 30 maj 2021
+ *      Author: Jakub
+ */
+
+#include <CurrentMeasurement.h>
+
+CurrentMeasurement::CurrentMeasurement(ADC_HandleTypeDef *hadc, DMA_HandleTypeDef *hdma) {
+	adcHandler = hadc;
+	dmaHandler = hdma;
+}
+
+void CurrentMeasurement::Init()
+{
+	lastNDTR = dmaHandler->Instance->NDTR;
+}
+
+
+uint16_t CurrentMeasurement::getCurrentData(uint8_t *dataBuffer)
+{
+	uint32_t currentNDTR = dmaHandler->Instance->NDTR;
+	uint32_t len = 0;
+
+	if (currentNDTR % 2 != 0) currentNDTR++; // I want to have got data pairs and also do not go outside the memory

```

Comment:
Consider making function that will have name according to comment that is here so the comment will be redundant.

Reply:
Changed in last commit, please check it @JacekJurek

Reply:
Looks good now